ui/src/app/views/account-creation/component.tsx

```diff
@@ -0,0 +1,312 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import {fullUrl, handleErrors} from 'app/utils/fetch';
+
+import {
+  CreateAccountRequest,
+  DataAccessLevel,
+  FetchArgs,
+  Profile,
+  ProfileApiFetchParamCreator,
+} from 'generated/fetch/api';
+
+import {
+  Error,
+  ErrorMessage,
+  LongInput,
+  styles as inputStyles
+} from 'app/components/inputs';
+
+import {
+  InfoIcon,
+  ValidationIcon
+} from 'app/components/icons';
+
+import {
+  TooltipTrigger
+} from 'app/components/popups';
+
+import {Button} from 'app/components/buttons';
+import { FormSection } from 'app/components/forms';
+import {BoldHeader} from 'app/components/headers';
+
+function isBlank(s: string) {
+  return (!s || /^\s*$/.test(s));
+}
+
+export interface AccountCreationProps {
+  invitationKey: string;
+  setProfile: Function;
+  onAccountCreation: Function;
+}
+
+export interface AccountCreationState {
+  profile: Profile;
+  usernameCheckInProgress: boolean;
+  usernameConflictError: boolean;
+  creatingAccount: boolean;
+  showAllFieldsRequiredError: boolean;
+}
+
+export class AccountCreationReact extends
+    React.Component<AccountCreationProps, AccountCreationState> {
+  usernameCheckTimeout: NodeJS.Timer;
+  accountCreated = false;
```

Comment:
Placing this boolean directly on `this` makes it 'stateful', but React can't see it and so it won't cause re-renders on change. Since the parent will end up unmounting this component on completion anyway, we can probably remove this.